src/main/kotlin/coverosR3z/persistence/PureMemoryDatabase.kt

```diff
@@ -48,7 +48,7 @@ class PureMemoryDatabase {
 
     fun addNewUser(userName: UserName, hash: Hash, salt: String) : Int {
         val newIndex = users.size + 1
-        users.add(User(newIndex, userName.value, hash, salt, null))
+        users.add(User(newIndex, userName.value, hash, salt, newIndex))
```

Comment:
Debatable.  We should discuss this.

Reply:
I think this requires more consensus on the design decision.

Reply:
Yeah, this is the simplest method of getting a unique id for them, which we need for this BDD case, but there are other options.

Reply:
Gone back to null